crates/resolver-tests/src/sat.rs

```diff
@@ -59,40 +61,130 @@ fn sat_at_most_one_by_key<K: std::hash::Hash + Eq>(
     by_keys
 }
 
-fn find_compatible_dep_summaries_by_name_in_toml(
+type DependencyVarMap<'a> =
+    HashMap<InternedString, HashMap<(DepKind, Option<&'a Platform>), varisat::Var>>;
+
+type DependencyFeatureVarMap<'a> = HashMap<
+    InternedString,
+    HashMap<(DepKind, Option<&'a Platform>), HashMap<InternedString, varisat::Var>>,
+>;
+
+fn create_dependencies_vars<'a>(
+    solver: &mut varisat::Solver<'_>,
+    pkg_var: varisat::Var,
+    pkg_dependencies: &'a [Dependency],
+    pkg_features: &FeatureMap,
+) -> (DependencyVarMap<'a>, DependencyFeatureVarMap<'a>) {
+    let mut var_for_is_dependencies_used = HashMap::<_, HashMap<_, _>>::new();
+    let mut var_for_is_dependencies_features_used = HashMap::<_, HashMap<_, HashMap<_, _>>>::new();
```

Comment:
Nit: these can use the type aliases you've declared above. `DependencyVarMap::new()`